pydm/widgets/embedded_display.py

```diff
@@ -176,16 +187,17 @@ def embedded_widget(self, new_widget):
             self.layout.removeWidget(self._embedded_widget)
             self._embedded_widget.deleteLater()
             self._embedded_widget = None
-        self._embedded_widget = new_widget
-        self._embedded_widget.setParent(self)
-        self.layout.addWidget(self._embedded_widget)
-        self.err_label.hide()
-        self._embedded_widget.show()
-        self._is_connected = True
+        if new_widget is not None:
```

Comment:
I think instead of watching for `open_file` to return `None`, I think it might be a cleaner pattern to only call `open_file` when you need to.